mlx/ops.cpp

```diff
@@ -83,10 +83,24 @@ array arange(
   if (std::isnan(start) || std::isnan(step) || std::isnan(stop)) {
     throw std::invalid_argument("[arange] Cannot compute length.");
   }
+
+  if (std::isinf(start) || std::isinf(stop)) {
+    throw std::invalid_argument("[arange] Cannot compute length.");
+  }
+
+  // Check if start and stop specify a valid range because if not, we have to
+  // return an empty array
+  if (std::isinf(step) &&
+      (step > 0 && start < stop || step < 0 && start > stop)) {
+    return array({start}, dtype);
+  }
```

Comment:
Another option that might simplify this is to do something like:
```suggestion
  if (std::isinf(step)) {
    step = (step < 0) * std::numeric_limits<double>::max();
  }
```

I'm not 100% certain, but I think the rest of the logic would flow unchanged after that.

Reply:
But wouldn't this be 0 when step is +inf?

Reply:
Right sorry! That is bug it should be `sign(step)` not `step<0`.  Or `(step < 0 ? -1 : 1)`

Reply:
This produces NaNs for some reason for the input start = 0, stop = 10, step = inf:
Currently I do this:
if (std::isinf(step)) {
    step = step > 0 ? std::numeric_limits<double>::max()
                    : std::numeric_limits<double>::min();
  }
  
 and then real_size is 1 so it should output an array with only the start but I guess due to some handling in the Arange primitive  it can't handle step being the numerical_limit because when I put 10000000 and -10000000 it works correctly.

Reply:
Ok no worries, I think it's fine the way you have it. Thanks for testing that!